### ISSUE TITLE
kubernetes.core.k8s_drain module "wait_for_completion" Parameter

### ISSUE THREAD

**Issue by itsek** [ansible-collections/kubernetes.core#593]
##### SUMMARY
new parameter for this module "wait_for_completion" (of drain)

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
kubernetes.core.k8s_drain module 

##### ADDITIONAL INFORMATION
currently, if you would issue something like this:

```
        - name: drain node
          kubernetes.core.k8s_drain:
            state: drain
            delete_options:
              terminate_grace_period: 300

        - name: reboot host
          reboot:

```

the following would happen: drain API call would be made with a grace prerioud for the pods of 300seconds, but since the ansible task "drain node" is complete - the reboot host task would be run immediatly after

so you have to do some inbetween checking yourself:

```
        - name: drain node
          kubernetes.core.k8s_drain:
            state: drain
            delete_options:
              terminate_grace_period: 300

       - name: wait for all pods to terminate
         until: all pods terminated
            < some magic depending on your pods / apps e.g. counting terminating pods >

       # this task runs only AFTER all pods have gracefully been terminated
        - name: reboot host
          reboot:
```

nice to have would be:
```
        - name: drain node
          kubernetes.core.k8s_drain:
            wait_for_completion: true
            state: drain
            delete_options:
              terminate_grace_period: 300

       # this task runs only AFTER all pods have gracefully been terminated, since k8s_drain "waits_for_completion"
        - name: reboot host
          reboot:
```

**Comment by gravesm** [comment]
The `wait_timeout` suboption of `delete_options` should do this, but I just tested locally and it looks like there's at least one bug with the wait logic. It will not wait at all if there is only one pod on the node. It does seem to work correctly if there are multiple pods, though.

**Comment by itsek** [comment]
thank your for the hint and the check, in the meantime for anyone looking for a workaround: 
calling kubectl with timeout and grace (timeout should be greater then grace) works fine, but might not be that elegant
```
        - name: drain node
          delegate_to: localhost
          ansible.builtin.command: "kubectl drain --timeout=330s --grace-period=300 --ignore-daemonsets --delete-local-data {{ inventory_hostname }}"
```

**Comment by beeankha** [comment]
> The `wait_timeout` suboption of `delete_options` should do this, but I just tested locally and it looks like there's at least one bug with the wait logic. It will not wait at all if there is only one pod on the node. It does seem to work correctly if there are multiple pods, though.

It's possible that there may be specific conditions that trigger the bug that aren't present in my environment (perhaps related to my Kubernetes version, cluster configuration, or pod setup), _or_ the issue might be intermittent / dependent on specific timing conditions, but I couldn't recreate this bug despite using several different versions of the `kubernetes.core` collection (`5.2.0`, `5.1.0`, `3.0.0`, `2.4.2`). As far as I could tell, the `wait_timeout` suboption seems to be functional even for a single pod on a node.

Since I couldn't recreate this bug, I'm going to go ahead and close this ticket; @itsek if you think that these suboptions are not sufficient and/or you are encountering the `wait_timeout` bug for a single pod, please feel free to re-open this ticket!
